src/DynamoPackages/PackageLoader.cs

```diff
@@ -421,19 +421,48 @@ public void LoadCustomNodesAndPackages(IEnumerable<string> newPaths, IPreference
             }
         }
 
-        [Obsolete("Do not use. This method is deprecated and will be removed in Dynamo 3.0. Use LoadCustomNodesAndPackages(IEnumerable<string> newPaths, IPreferences preferences, CustomNodeManager customNodeManager) instead.")]
         /// <summary>
-        /// This method is called when custom nodes and packages need to be reloaded if there are new package paths.
+        /// To load custom nodes and packages from temporary custom paths,
+        /// initialize a local PreferenceSettings object and add the paths to its CustomPackageFolders property,
+        /// then initialize LoadPackageParams with this preferences object and use as input to this method.
+        /// To load from custom paths that need to be persisted to the preferences, 
+        /// initialize a LoadPackageParams from an existing preferences object.
         /// </summary>
-        /// <param name="loadPackageParams"></param>
+        /// <param name="loadPackageParams">LoadPackageParams initialized with local PreferenceSettings object containing custom package path.</param>
         /// <param name="customNodeManager"></param>
         public void LoadCustomNodesAndPackages(LoadPackageParams loadPackageParams, CustomNodeManager customNodeManager)
         {
             foreach (var path in loadPackageParams.Preferences.CustomPackageFolders)
             {
                 customNodeManager.AddUninitializedCustomNodesInPath(path, false, false);
+                if (!pathManager.PackagesDirectories.Contains(path))
+                {
+                    if (DynamoModel.IsDisabledPath(path, loadPackageParams.Preferences))
+                    {
+                        Log(string.Format(Resources.PackagesDirectorySkipped, path));
+                        continue;
+                    }
+                    else
+                    {
+                        ScanPackageDirectories(path, loadPackageParams.Preferences);
+                    }
+                }
+            }
+            if (pathManager != null)
+            {
+                foreach (var pkg in LocalPackages)
+                {
+                    if (Directory.Exists(pkg.BinaryDirectory))
+                    {
+                        pathManager.AddResolutionPath(pkg.BinaryDirectory);
+                    }
+                }
+            }
+
+            if (LocalPackages.Any())
+            {
+                LoadPackages(LocalPackages);
```

Comment:
looks like this might load existing packages too

Reply:
The other one does the following
```
            // Load only those recently addeed local packages (that are located in any of the new paths)
                var newPackages = LocalPackages.Where(x => newPaths.Any(y => x.RootDirectory.Contains(y)));
                LoadPackages(newPackages);
 ```

Reply:
My intention was to leave the original API unchanged (in behavior) so as not to affect third-party devs.

Reply:
Well you already modified the function body ... so the behavior is changed

Reply:
The logic is the same as before.

Reply:
I have only refactored the method.